corehq/apps/userreports/templates/userreports/reportbuilder/configure_report.html

```diff
@@ -311,9 +320,17 @@ <h3 data-bind="visible: !previewError()">{% trans 'Data Table' %}</h3>
             "initialLocation": {{ initial_location|JSON }},
             "initialChartType": {{ initial_chart_type|JSON }},
             "mapboxAccessToken": {{ MAPBOX_ACCESS_TOKEN|JSON }},
-            "hasReportBuilderAccess": {{ has_report_builder_access|JSON }},
+            // In previewMode, report can't be saved.
+            "previewMode":
+              {% if not has_report_builder_access or at_report_limit and not existing_report%}
```

Comment:
yeah I struggled with how to make this readable. Annoyingly, django doesn't allow parentheses in template tags. These pieces of page context get computed in a few different functions, so it's not super easy to just do this logic in the view either.
I think I'll add a comment though that shows the logic with parens.